packages/core/src/lib/base_protocol.ts

```diff
@@ -89,4 +93,15 @@ export class BaseProtocol implements IBaseProtocol {
     // Filter the peers based on the specified criteria
     return filterPeers(allPeersForProtocol, numPeers, maxBootstrapPeers);
   }
+
+  protected ensurePubsubTopicIsValid(
```

Comment:
This is better suited as a utility function as it has no dependencies on a particular protocol.